components/script/dom/urlhelper.rs

```diff
@@ -43,6 +43,25 @@ impl UrlHelper {
         let _ = wrapper.set_host(&value.0);
     }
 
+    pub fn Origin(url: &Url) -> USVString {
+        USVString(match url.origin() {
+            Origin::UID(_) => {
+                // https://html.spec.whatwg.org/multipage/#unicode-serialisation-of-an-origin
+                // If the origin in question is not a scheme/host/port tuple,
+                // then return the literal string "null" and abort these steps.
+                "null".to_owned()
+            },
+            Origin::Tuple(protocol, host, port) => {
+                format!(
+                    "{protocol}://{host}:{port}",
```

Comment:
We need to be smarter here according to step 5 of https://html.spec.whatwg.org/multipage/browsers.html#unicode-serialisation-of-an-origin .